### PR TITLE
Done

### PR DESCRIPTION
**Pull request by barbanihor**
DEMO LINK](https://Pidsvynok.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,79 @@
...
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [select, setSelect] = useState('all');
+  const [query, setQuery] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => setTodos(todosFromServer))
```
Comment (volodymyr-soltys97) [inline src/App.tsx:27]:
```suggestion
      .then(setTodos)
```

File: src/components/TodoFilter/TodoFilter.tsx [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
       <span className="select">
-        <select data-cy="statusSelect">
+        <select
+          value={select}
+          onChange={e => onSelect(e.target.value)}
+          data-cy="statusSelect"
+        >
           <option value="all">All</option>
           <option value="active">Active</option>
           <option value="completed">Completed</option>
```
Comment (volodymyr-soltys97) [inline src/components/TodoFilter/TodoFilter.tsx:26]: Create a enum for 'all', 'active', 'completed' and use it here

File: src/components/TodoFilter/TodoFilter.tsx [hunk 3]
```diff
@@ -23,7 +43,14 @@ export const TodoFilter = () => (
 
       <span className="icon is-right" style={{ pointerEvents: 'all' }}>
         {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
+        {query && (
+          <button
+            onClick={() => clearInput()}
```
Comment (volodymyr-soltys97) [inline src/components/TodoFilter/TodoFilter.tsx:48]:
```suggestion
            onClick={clearInput}
```

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -1,100 +1,115 @@
...
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
+      <tbody>
+        {todos.map(todo => (
+          <tr key={todo.id} data-cy="todo" className="">
```
Comment (volodymyr-soltys97) [inline src/components/TodoList/TodoList.tsx:29]:
```suggestion
          <tr key={todo.id} data-cy="todo">
```

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -1,100 +1,115 @@
...
+              {todo.completed && (
+                <span className="icon" data-cy="iconCompleted">
+                  <i className="fas fa-check" />
+                </span>
+              )}
+            </td>
+            <td className="is-vcentered is-expanded">
+              <p
+                className={
+                  todo.completed ? 'has-text-success' : 'has-text-danger'
```
Comment (volodymyr-soltys97) [inline src/components/TodoList/TodoList.tsx:41]: Use the classnames library rules for add classes with condition

File: src/components/TodoList/TodoList.tsx [hunk 6]
```diff
@@ -1,100 +1,115 @@
...
+              </button>
+            </td>
+          </tr>
+        ))}
+        {/* <tr data-cy="todo" className="">
+          <td className="is-vcentered">1</td>
+          <td className="is-vcentered" />
+          <td className="is-vcentered is-expanded">
+            <p className="has-text-danger">delectus aut autem</p>
+          </td>
```
Comment (volodymyr-soltys97) [inline src/components/TodoList/TodoList.tsx:71]: Remove all comments

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code
You need to add data-cy attribute for tests 
image

File: src/components/TodoList/TodoList.tsx (hunk 5 above)
Comment (Pidsvynok) [reply src/components/TodoList/TodoList.tsx:41]: <p className={classNames({
                'has-text-success': todo.completed,
                'has-text-danger': !todo.completed,
              })}
              
              Like this?

Comment (Pidsvynok) [reply src/components/TodoList/TodoList.tsx:41]: <p className={classNames({
                'has-text-success': todo.completed,
                'has-text-danger': !todo.completed,
              })}

Comment (Pidsvynok) [reply src/components/TodoList/TodoList.tsx:41]: className={classNames({
                'has-text-success': todo.completed,
                'has-text-danger': !todo.completed,
              })}

**Comment by Pidsvynok** [comment]
Fixed, except for data-cy, to be honest, I'm not really sure where to put it, because I tried to put it everywhere I could

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Feel free to ask for help in the chat to pass tests

**Pull request closed without merging**